webapp/src/js/controllers/inbox.js

```diff
@@ -100,56 +100,66 @@ var _ = require('underscore'),
       document.title = title;
     });
 
+    const SYNC_STATUS = {
+      inProgress: {
+        icon: 'fa-refresh',
+        key: 'sync.status.in_progress',
+        debounce: true
+      },
+      success: {
+        icon: 'fa-check',
+        key: 'sync.status.not_required',
+        className: 'success'
+      },
+      required: {
+        icon: 'fa-exclamation-triangle',
+        key: 'sync.status.required',
+        className: 'required'
+      },
+      unknown: {
+        icon: 'fa-question-circle',
+        key: 'sync.status.unknown'
+      }
+    };
+
     $scope.replicationStatus = {
       disabled: false,
       lastSuccess: {},
       lastTrigger: undefined,
-      current: 'unknown',
-      textKey: 'sync.status.unknown',
-    };
-    var SYNC_ICON = {
-      in_progress: 'fa-refresh',
-      not_required: 'fa-check',
-      required: 'fa-exclamation-triangle',
-      unknown: 'fa-question-circle',
-    };
-
-    const updateReplicationStatus = status => {
-      if ($scope.replicationStatus.current !== status) {
-        $scope.replicationStatus.current = status;
-        $scope.replicationStatus.textKey = 'sync.status.' + status;
-        $scope.replicationStatus.icon = SYNC_ICON[status];
-      }
+      current: SYNC_STATUS.unknown,
     };
 
-    DBSync.addUpdateListener(update => {
-      if (update.disabled) {
+    DBSync.addUpdateListener(({ disabled, unknown, inProgress, to, from }) => {
```

Comment:
I was trying to avoid passing around magic strings, but these property names are obviously just magic strings too, so I guess it makes little difference. I'll make the change.

I think an event emitter pattern would be a little cleaner but it's still just magic strings. Eventually this will be stored in redux which probably makes the whole thing moot...

Reply:
I would personally prioritize clean public interfaces. Your goal to avoid magic strings and leverage compile-time validation seems achieveable with the [enum](https://google.github.io/styleguide/jsguide.html#features-objects-enums) pattern you already use in this pr.

Reply:
The property names at least have the advantage of compile-time verification if you mispell or use one that is undefined.  Yay redux!